### PR TITLE
Adds doc for wire vs protoc

### PR DESCRIPTION
**Pull request by oldergod**
Probably can add the basic implementation details we already discuss in the README later. As well as byte strings.

### CODE REVIEW

File: wire-library/docs/wire_vs_protoc.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    .build()
+val list: List =
+    ListValue.newBuilder().apply {
+        addValues(Value.newBuilder.setStringValue(“a”).build())
+        addValues(Value.newBuilder.setNumberValue(2.0).build())
+    }
+    .build()
+val boolValue = Value.newBuilder.setBoolValue(true).build()
+val nullValue = Value.newBuilder().setNullValue(NullValue.NULL_VALUE)
+        .build()
```
Comment (Egorand) [inline wire-library/docs/wire_vs_protoc.md:108]: nit: indentation

File: wire-library/docs/wire_vs_protoc.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// Wire kotlin
+val floatValue = 33f
+
+// protoc
+val floatValue = FloatValue.newBuilder().setValue(33f).build()
+```
+
+### JSON
+
+While Proto2 didn’t, Proto3 defines Protobuf serialization over JSON. Wire and protoc are
```
Comment (Egorand) [inline wire-library/docs/wire_vs_protoc.md:127]: nit: the capitalization in `proto3` (vs `Proto3`) is inconsistent across the doc, better use one or the other everywhere

**Review by Egorand — APPROVED** [review]
Nice!

File: wire-library/docs/wire_vs_protoc.md (hunk 1 above)
Comment (oldergod) [reply wire-library/docs/wire_vs_protoc.md]: fixed

File: wire-library/docs/wire_vs_protoc.md (hunk 2 above)
Comment (oldergod) [reply wire-library/docs/wire_vs_protoc.md]: fixed

File: wire-library/docs/wire_vs_protoc.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+// Protoc
+val any: Any = Any.pack(foo)
+val person: Person = any.unpack(Person.class)
+```
+
+### Duration & Timestamp
+
+Both `google.protobuf.Duration` and `google.protobuf.Timestamp` types will be generated by using
+their JVM equivalent: `java.time.Duration` and `java.time.Instant`. For non-JVM language, we provide
```
Comment (swankjesse) [inline wire-library/docs/wire_vs_protoc.md:39]: non-JVM language -> non-JVM platforms

File: wire-library/docs/wire_vs_protoc.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+val floatValue = 33f
+
+// Protoc
+val floatValue = FloatValue.newBuilder().setValue(33f).build()
+```
+
+### JSON
+
+While Proto2 didn’t, Proto3 defines Protobuf serialization over JSON. Wire and Protoc are
+inter-compatible but their API are quite different. Wire offers JSON serialization over Moshi or
```
Comment (swankjesse) [inline wire-library/docs/wire_vs_protoc.md:127]: Link `Moshi` to https://github.com/square/moshi/
Link `Gson` to https://github.com/google/gson/

**Review by swankjesse — APPROVED** [review]

**Pull request merged**